pkg/logs/input/file/file_provider.go

```diff
@@ -130,6 +131,20 @@ func (p *Provider) searchFiles(pattern string, source *config.LogSource) ([]*Fil
 		return nil, fmt.Errorf("could not find any file matching pattern %s, check that all its subdirectories are executable", pattern)
 	}
 	var files []*File
+
+	// Files are sorted because of a heuristic on the filename: often the filename and/or the folder name
+	// contains information in the file datetime. Most of the time we want the most recent files.
+	// Here, we reverse paths to have stable sort keep reverse lexicographical order w.r.t dir names. Example:
+	// [/tmp/1/2017.log, /tmp/1/2018.log, /tmp/2/2018.log] becomes [/tmp/2/2018.log, /tmp/1/2018.log, /tmp/1/2017.log]
+	// then kept as is by the sort below.
+	for i := len(paths)/2 - 1; i >= 0; i-- {
```

Comment:
can you link to https://github.com/golang/go/wiki/SliceTricks#reversing in a comment ?